components/src/core/Drawer/DrawerRailItem.tsx

```diff
@@ -24,48 +30,36 @@ export type DrawerRailItemClasses = {
     ripple?: string;
 };
 
-export type DrawerRailItemProps = MuiButtonBaseProps &
-    Pick<
-        PXBlueDrawerInheritableProperties,
-        | 'activeItemBackgroundColor'
-        | 'activeItemFontColor'
-        | 'activeItemIconColor'
-        | 'divider'
-        | 'itemFontColor'
-        | 'itemIconColor'
-        | 'onItemSelect'
-        | 'ripple'
-    > & {
-        activeItem?: string;
+export type ExtendedNavItem = NavItem & { ButtonBaseProps?: Partial<MuiButtonBaseProps> };
```

Comment:
Right now if I try to do 
```tsx
<DrawerNavGroup items={[{ icon: <></>, itemID: 'asdf' }]} />
```
it will fail, but if I do 
```tsx
<DrawerNavGroup>
    <DrawerRailItem itemID={''} icon={<></>} />
</DrawerNavGroup>
```
It will work.

Maybe we can do some console warning tricks, or insert a blurp about this in the readme.

Reply:
I updated the type for items to take an array of navItems or railItems, so you should be able to do either of these forms now.

Reply:
It gets cast to the InfoListItem level and I am still getting yelled at. I was expecting:

### Pass
```tsx
        <Drawer variant={'rail'}>
            <DrawerBody>
                <DrawerNavGroup items={[{ icon: <></>, itemID: 'b' }]} />
                <DrawerNavGroup>
                    <DrawerRailItem icon={<></>} itemID={'a'} />
                </DrawerNavGroup>
            </DrawerBody>
        </Drawer>
```

### Fail (or error thrown)
```tsx
        <Drawer variant={/* some other variant */}>
            <DrawerBody>
                <DrawerNavGroup items={[{ icon: <></>, itemID: 'b' }]} />
                <DrawerNavGroup>
                    <DrawerRailItem icon={<></>} itemID={'a'} />
                </DrawerNavGroup>
            </DrawerBody>
        </Drawer>
```

### Pass
```tsx
        <Drawer variant={/* some other variant */}>
            <DrawerBody>
                <DrawerNavGroup items={[{ title: 'b', itemID: 'b' }]} />
                <DrawerNavGroup>
                    <DrawerRailItem title={'a'} itemID={'a'} />
                </DrawerNavGroup>
            </DrawerBody>
        </Drawer>
```

### Fail (or error thrown)
```tsx
        <Drawer variant={'rail'}>
            <DrawerBody>
                <DrawerNavGroup items={[{ title: 'b', itemID: 'b' }]} />
                <DrawerNavGroup>
                    <DrawerRailItem title={'a'} itemID={'a'} />
                </DrawerNavGroup>
            </DrawerBody>
        </Drawer>
```

Reply:
Fixing number 1 (it's just a small issue with the PropTypes definition). Numbers 2 and 3 are not valid use cases...you shouldn't be using a RailItem inside of a drawer that is not the rail variant...i don't see us supporting that.

4 is failing because you just told me to make the icon prop required for a DrawerRailItem and you haven't provided one.

Reply:
I can go back to making the icon optional if that's what we want to do...the rail item can functionally handle it either way.

Reply:
>  RailItem inside of a drawer that is not the rail variant

My bad for copy pasting. Corrected

Reply:
Okay, so 2 is failing because you don't have a `title` prop which is required for NavItems. And 4 is failing because you don't have an `icon`, which is required for a RailItem.

Reply:
Yup, that's the expected behavior. I see that you made more changes... will review during the OH